### PR TITLE
Improve precision of matched bytes in lookahead

### PR DESCRIPTION
**Pull request by brendanzab**
The records all possible bytes matched in a hash set, letting us decode the full start-of-frame alternation. This could be made more efficient, but works for now.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -1,10 +1,66 @@
...
+                HashSet::is_disjoint(included0, included1)
+            }
+            // If the set of contained bytes are a subset of the excluded bytes,
+            // then the byte sets are disjoint
+            (ByteSet::Includes(included), ByteSet::Excludes(excluded))
+            | (ByteSet::Excludes(excluded), ByteSet::Includes(included)) => {
+                HashSet::is_subset(included, excluded)
+            }
+            // Too hard to enumerate all of these: just say that they aren’t disjoint
+            (ByteSet::Excludes(_), ByteSet::Excludes(_)) => false,
```
Comment (mikeday) [inline src/main.rs:43]: Is there any reason not to check [0..255] ? Seems like it might be a one liner :thinking:

Comment (brendanzab) [reply src/main.rs]: Not sure if I got it right!

File: src/main.rs [hunk 2]
```diff
@@ -40,8 +40,11 @@ impl ByteSet {
             | (ByteSet::Excludes(excluded), ByteSet::Includes(included)) => {
                 HashSet::is_subset(included, excluded)
             }
-            // Too hard to enumerate all of these: just say that they aren’t disjoint
-            (ByteSet::Excludes(_), ByteSet::Excludes(_)) => false,
+            // Hard: enumerate these by brute force - they are disjoint if all
+            // bytes are contained exclusively in one byte set or the other set
```
Comment (mikeday) [inline src/main.rs:44]: Does it need to be exclusive? I think they can be in both, they just have to be in at least one :thinking:

Comment (mikeday) [reply src/main.rs:44]: Right, in order for two bytesets to be disjoint there must be one byte in both sets, which means there must be one byte that isn't excluded by _either_ set, but a byte excluded by _both_ sets is fine.

**Comment by brendanzab** [comment]
Think this should be ready!

**Review by mikeday — APPROVED** [review]
Looks good to me!

**Pull request merged**